op-tests/test-blspy-pairing.txt

```diff
@@ -0,0 +1,32 @@
+; This file was generated by tools/generate-bls-tests.py
+
+bls_pairing_identity ((0xa4a108e5dabd6d6fde629fe9515cf397ab2a2d44c5740d39525aa7f5230df38d6753cb7a0683a75e2f4f3d82fd46cdb0 . 0xa9c8ab9f27d84f764c709c82c516e5d6782d43751c55a48cb39999e321f3a896696878838608b3764c43e20af5ab1c2e14cf7a274ebd8f9f418a050bab94ffd44d7a054c41b08425fdf0c7b2d1317cd5cf121749f456068b113ef606aebacd2c) (0xb66837657cd1564bb04e9acb1efd96b53ef90502bb000e6ca44da7e07f655e6fb50606f3cc47fd936ba4108ccf592625 . 0xb4e0d25a3588446f00043183bd3e072eb50765ce3f57ec7cca1f10014117b33530aec7ceff37c291d267a66c557f49d000d255b10ca4804d0bebcec9cb073577eaa416673f9f3bc6bf2936c0be7154fc1cccc430003aaae5311f0d67673942d3) (0x8f3eb0208a3f0808df538ebc1f582d01c62baabe12842171f7d6c01f9b31f7c9d386190d9349d53378998d0162c7ecb2 . 0x8bb781464f4a21418284f8307755e88a696585b56edbfe3c927998e66aea3fccaded0830ef3500f2c8a46bf16bff7dbf1388b511109da614b928f0d17b8e8d7d0cded6dc2477e9c3c3cd5c97b41067ed3200cd7016eff07e45b79112cc1d6a13) (0xb066ce9e6484fe574ed55003449fae4d08652f85de45d718c2397c667cd80ed5571458d58c7d28b101ab1ef986973e77 . 0x8962435666bd7d6d6a94210e0f52f45d0b516139e4e9925bad4f0bf3e96dc6c62db5e50d9f86c1cef5e34caa79f4585e036a43550b14f0269bdbdbcee627b004613c96a7f4195970a0084214a1c2a40decf032efb94e657005a0da78c77d3d0d) (0xa73e3a434a117bed5c0eab24dce24b2e03bc75254b878900eb01f98cf389329cdd7d83cd8033475ff1ec6b2e10abbb09 . 0xa7ce0b706c4b6de43444ce02353bbee0d953fe403880fb49668ac1244bad2eeb5e3f83f3f526676d3513f56e32c7dfdd143061616b34d784edcde312c5b2d024878cb3b965895660d6727f5c210f73f60ed91be68b5008ad09048585c74950df) (0xb756ecad886b575d2c95014c20bcd9b52f1f72e3b5fbc657bacf1e72be707f9aaf76a9156a1628b99d873036beecec10 . 0xaa898708a47aa64d7d2d1d5d5b80d5609bb6dd088f8d40c3a82983fa517878a748c54acd162176871c12b404073a8f4119a7f4d04779af2b2d59949fa5d80c57bd8ed4262e4c186953ec28ecc69a02789474816f62b40184a448fcaaba7b67b6) (0xae0154514ac83b8efea5b2eee425fc41ead032bb04b8a977eff85e80f9dd8940833c0d7d85921bf18fe84f639f417cca . 0xa63a338eb15d6b1f3c4d6b6989f851841306dc93bad8963176a8c3e10af3de84254905d9bdb0afbe2a34cb2e3d98e4a6020ae15384e0a35b36658a10fb3c2d7391a545019acbc6892cbf9447628be9852e788f31e7be1078937102eb976bbac4) (0xb7f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb . 0x8ded853eb71f6015bc2c338cc2e2f2540e1d4cdc7d662965cd8827d6482e321e94ebd2cd299f77063336a531e294d879047d7017b93b232ce95a26e52cef34a1120aa4a3ab256242044d33e15b5430ff58612102bc80ef195a3adb4477cb7e1b) ) => 1 | 41122591
```

Comment:
I'm not convinced that this format makes the most sense. I think need to actually implement a proof to see. If our loops produce N values from g1 and N values from g2, to call this, we'd need to do a `zip` operation in clvm which might better be done in the vm implementation. Compare to `(bls_pairing_identity g1_list g2_list)`. I don't know it's better, which is why I think we need to implement the toy to see.

Reply:
good point. my impression so far is that this list is unlikely to be dynamic. in the snarkjs case, it's hard-coded to be 4 pairs

Reply:
here's the equivalent call in snarkjs' proof verification: https://github.com/iden3/snarkjs/blob/master/src/groth16_verify.js#L62-L68


It has a fixed number of arguments.
```
    const res = await curve.pairingEq(
        curve.G1.neg(pi_a) , pi_b,
        cpub , vk_gamma_2,
        pi_c , vk_delta_2,
        vk_alpha_1, vk_beta_2
    );
```

Reply:
If this is typical, then yes, varargs makes the most sense.